packages/openapi-to-graphql/src/oas_3_tools.ts

```diff
@@ -406,6 +406,13 @@ export function getSchemaTargetGraphQLType<TSource, TContext, TArgs>(
   schema: SchemaObject,
   data: PreprocessingData<TSource, TContext, TArgs>
 ): string | null {
+  if (schema.oneOf) {
```

Comment:
Unfortunately, I don't think we can do this as `oneOf` and `allOf` does not necessitate union or object. 

For example, this is an [example](https://json-schema.org/understanding-json-schema/reference/combining.html) provided by `json-schema`:

```
{
   "type": "number",
   "oneOf": [
     { "multipleOf": 5 },
     { "multipleOf": 3 }
   ]
 }
```

***

Additionally, even if `oneOf` contains a number of schemas, we still need to make a number of checks to see if we can still create a union as JSON schema `oneOf` is a lot more flexible than GraphQL `union type`. 

For example, according to the [specification](https://spec.graphql.org/June2018/#sec-Unions), GraphQL unions cannot be composed of other unions

> The member types of a Union type must all be Object base types; Scalar, Interface and Union types must not be member types of a Union. Similarly, wrapping types must not be member types of a Union.

That is why we have this [check](https://github.com/IBM/openapi-to-graphql/blob/092b0d50be156f97b12e20e295b48aec5770c637/packages/openapi-to-graphql/src/preprocessor.ts#L764), to see if we have cases of nested `oneOf`/`allOf`.